grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx

```diff
@@ -32,481 +32,613 @@ import { IntegrationHelper } from 'pages/integration/Integration.helper';
+    selectedIntegration,
+    onSubmit,
+    onHide,
+    onBackClick,
+  }: IntegrationFormProps) => {
+    const store = useStore();
+    const history = useHistory();
+    const styles = useStyles2(getIntegrationFormStyles);
+    const isNew = id === 'new';
+    const { userStore, grafanaTeamStore, alertReceiveChannelStore } = store;
+
+    const data = isNew
+      ? { integration: selectedIntegration?.value, team: userStore.currentUser?.current_team, labels: [] }
       : prepareForEdit(alertReceiveChannelStore.items[id]);
 
-  const { alertReceiveChannelOptions } = alertReceiveChannelStore;
-
-  const options = alertReceiveChannelOptions
-    ? alertReceiveChannelOptions.filter((option: ApiSchemas['AlertReceiveChannelIntegrationOptions']) => {
-        if (option.value === 'grafana_alerting' && !window.grafanaBootData.settings.unifiedAlertingEnabled) {
-          return false;
-        }
-
-        // don't allow creating direct paging integrations
-        if (option.value === 'direct_paging') {
-          return false;
-        }
-
-        return (
-          option.display_name.toLowerCase().includes(filterValue.toLowerCase()) &&
-          !option.value.toLowerCase().startsWith('legacy_')
-        );
-      })
-    : [];
-
-  const extraGFormProps: { customFieldSectionRenderer?: React.FC<CustomFieldSectionRendererProps> } = {};
+    const { integration } = data;
+
+    const formMethods = useForm<FormFields>({
+      defaultValues: isNew
+        ? {
+            // these are the default values for creating an integration
+            [FormFieldKeys.Integration]: integration,
+            [FormFieldKeys.DefaultWebhooks]: true,
+          }
+        : {
+            // existing values from existing integration (edit-mode)
+            ...data,
+          },
+      mode: 'all',
+    });
+
+    const {
+      control,
+      handleSubmit,
+      formState: { errors },
+    } = formMethods;
+
+    const [
+      {
+        isExistingContactPoint,
+        dataSources,
+        contactPoints,
+        selectedAlertManagerOption,
+        selectedContactPointOption,
+        allContactPoints,
+      },
+      setState,
+    ] = useReducer(
+      (state: GrafanaContactPointState, newState: Partial<GrafanaContactPointState>) => ({
+        ...state,
+        ...newState,
+      }),
+      {
+        isExistingContactPoint: true,
+        selectedAlertManagerOption: undefined,
+        selectedContactPointOption: undefined,
+        dataSources: [],
+        contactPoints: [],
+        allContactPoints: [],
+      }
+    );
+
+    useEffect(() => {
+      (async function () {
+        setState({
+          allContactPoints: await AlertReceiveChannelHelper.getGrafanaAlertingContactPoints(),
+        });
+      })();
+    }, []);
+
+    const labelsRef = useRef(null);
+
+    const [labelsErrors, setLabelErrors] = useState([]);
+    const isServiceNow = IntegrationHelper.isSpecificIntegration(integration, 'servicenow');
```

Comment:
We already have `getIsBidirectionalIntegration` which is out sort of feature flag check to enable / disable things related to SNOW. It would be good to use a single source of truth for it (we can rename `getIsBidirectionalIntegration` if you want)